Userland/Utilities/nohup.cpp

```diff
@@ -0,0 +1,120 @@
+            exit(127);
+        }
+
+        path = LexicalPath::join(LexicalPath::canonicalized_path(home_env), path).string();
+        fd_or_error = Core::System::open(path, options, mode);
+        if (fd_or_error.is_error()) {
+            warnln("nohup: unable to open for appending given path {}", path);
+            exit(127);
+        }
+    }
+
+    fd = fd_or_error.value();
+
+    if (dup2(fd, fd_to_redirect) == -1) {
+        dbgln("nohup: redirection failed: {}", strerror(errno));
+        exit(127);
+    }
+
+    if (fd_to_redirect != STDERR_FILENO)
+        out(stderr, "appending output to {}\n", path);
+}
+
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio wpath cpath exec sigaction"));
+
+    DeprecatedString utility;
+    Vector<DeprecatedString> args;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_stop_on_first_non_option(true);
+    args_parser.set_general_help("Invoke a utility that will ignore SIGHUPs");
+    args_parser.add_positional_argument(utility, "Utiltiy to be invoked", "utility");
+    args_parser.add_positional_argument(args, "Arguments to pass to utility", "args", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    auto stdout_is_a_tty = false;
+    auto bool_or_error = Core::System::isatty(STDOUT_FILENO);
+    if (bool_or_error.is_error()) {
+        auto error = bool_or_error.release_error();
+        if (error.code() != EBADF) {
+            warnln("nohup: error while performing tty check on stdout: {}", strerror(error.code()));
+            return 127;
+        }
+    } else {
+        stdout_is_a_tty = bool_or_error.value();
+    }
+
+    if (stdout_is_a_tty)
+        dup_out_file(STDOUT_FILENO);
+
+    auto stderr_is_a_tty = false;
+    bool_or_error = Core::System::isatty(STDERR_FILENO);
+    if (bool_or_error.is_error()) {
+        auto error = bool_or_error.release_error();
+        if (error.code() != EBADF) {
+            warnln("nohup: error while performing tty check on stderr: {}", strerror(error.code()));
+            return 127;
+        }
+    } else {
+        stderr_is_a_tty = bool_or_error.value();
+    }
+
+    if (stderr_is_a_tty && dup2(STDOUT_FILENO, STDERR_FILENO) == -1) {
+        if (errno != EBADF) {
+            warnln("nohup: error redirecting stderr to stdout: {}", strerror(errno));
+            return 127;
+        }
+        // NOTE: Standard output must be closed, so "...the same output shall instead be
+        //       appended to the end of the nohup.out file..."
+        dup_out_file(STDERR_FILENO);
+    }
+
+    size_t size = args.size() + 2;
+    char* argv[size];
+
+    argv[0] = const_cast<char*>(utility.characters());
+
+    for (size_t i = 0; i < args.size(); i++) {
+        auto* arg = const_cast<char*>(args.at(i).characters());
+        argv[i + 1] = arg;
+    }
+
+    argv[size - 1] = NULL;
+
+    (void)signal(SIGHUP, SIG_IGN);
+
+    execvp(argv[0], argv);
```

Comment:
Could `TRY(Core::System::exec(utility, args, Core::System::SearchInPath::Yes));` be used here? The name of the command is expected to be included in `args`, I believe.

Reply:
I was able to use it, but I had to prepend `utility` to `args` since args doesn't include the command as well as pledge `rpath` since the call needs to use `Core::System::access()`.